plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -437,3 +478,7 @@ instance Closed DefaultUni where
     bring _ (f `DefaultUniApply` _ `DefaultUniApply` _ `DefaultUniApply` _) _ =
         noMoreTypeFunctions f
     bring _ DefaultUniData r = r
+    bring _ DefaultUniBLS12_381_G1_Element r = r
+    bring _ DefaultUniBLS12_381_G2_Element r = r
+    bring _ DefaultUniBLS12_381_MlResult  r = r
```

Comment:
It wasn't too bad, although the problem with the newtypes and the fact that the `MlResult` type is opaque and non-serialisable were unexpected and caused some difficulties.  Fortunately it turned out to be not too difficult to sort those out though.

Reply:
> It wasn't too bad, although the problem with the newtypes and the fact that the MlResult type is opaque and non-serialisable were unexpected and caused some difficulties. 

Yeah, it would be very hard for us to predict the latter. The former is unfortunate, but there's hope for fixing it.